### PR TITLE
attributeFields - Map option

### PR DESCRIPTION
**Pull request by bringking**
This PR implements the map option for attributeFields with an object hash and function. Included are two unit tests that cover using a function or using an object hash

### CODE REVIEW

File: src/attributeFields.js [hunk 1]
```diff
@@ -12,16 +12,26 @@ module.exports = function (Model, options) {
     var attribute = Model.rawAttributes[key]
       , type = attribute.type;
 
-    memo[key] = {
+    // determine the key
+    var mappedKey = key;
+    if (options.map) {
+      if (typeof options.map === 'function') {
+        mappedKey = options.map(key);
```
Comment (mickhansen) [inline src/attributeFields.js]: Might want a `|| key` here too, let users ignore those they don't want to map

File: src/attributeFields.js [hunk 2]
```diff
@@ -12,16 +12,26 @@ module.exports = function (Model, options) {
     var attribute = Model.rawAttributes[key]
       , type = attribute.type;
 
-    memo[key] = {
+    // determine the key
+    var mappedKey = key;
```
Comment (mickhansen) [inline src/attributeFields.js]: Couldn't we just overwrite key? I don't see it being used for anything else

Comment (bringking) [inline src/attributeFields.js]: Yeah I agree, not sure why I wrote it that way. I will fix it ASAP

**Comment by bringking** [comment]
Updated based on your comments, let me know if it looks okay

**Comment by mickhansen** [comment]
Looks good, can i ask you to add a documentation update aswell?

**Comment by bringking** [comment]
Absolutely, give me a few mins

**Comment by bringking** [comment]
FYI these docs are complete

**Pull request merged**